packages/server-wallet/src/chain-service/chain-service.ts

```diff
@@ -81,16 +81,19 @@ export class ChainService implements ChainServiceInterface {
 
   private finalizingChannels: {finalizesAtS: number; channelId: Bytes32}[] = [];
 
-  constructor({
-    provider,
-    pk,
-    pollingInterval,
-    logger,
-    blockConfirmations,
-    allowanceMode,
-  }: Partial<ChainServiceArgs>) {
+  constructor(
+    {
+      provider,
+      pk,
+      pollingInterval,
+      logger,
+      blockConfirmations,
```

Comment:
Wrt the shortcoming in your description, how about:

``` ts
constructor(
    {
      provider,
      pk,
      pollingInterval,
      logger,
      blockConfirmations,
      allowanceMode,
      chainNetworkId,
    }: Partial<ChainServiceArgs> & {chainNetworkId: number; pk: string} // chainNetworkId: number
  )
```
?

This makes the chainNetworkId and pk arguments required, avoiding the '2nd argument' problem and also the need for the `if (!pk) throw` check in the first line of the constructor (assuming this ctor doesn't have any remote invocation stuff that wouldn't know about the required pk arg).

Reply:
(Remote invocations seem unlikely for this ctor given that the thing I'm talking about is a private key.....)

Reply:
Thanks for this suggestion @NiloCK! I only went partway there, since the `pk` thing a) requires removing `undefined` from the type of `pk` in a few more places besides this (higher up in the stack) and b) is a slight mission creep. I did put a TODO in for it, though 73f48f6.